theano/compile/mode.py

```diff
@@ -111,6 +113,16 @@ class AddDestroyHandler(gof.Optimizer):
 
     """
     def apply(self, fgraph):
+        supervisor_added = False
+        for feature in fgraph._features:
+            if isinstance(feature, Supervisor):
+                supervisor_added = True
+                return
```

Comment:
Thanks, I will fix it but it wouldn't make any difference since the ```if``` would always be false when we returned.